### ISSUE TITLE
Notice: Undefined index: value in /vendor/divante/magento2-vsbridge-indexer/src/module-vsbridge-indexer-catalog/Model/Indexer/DataProvider/Product/ConfigurableData.php on line 254

### ISSUE THREAD

**Issue by janmyszkier** [DivanteLtd/magento2-vsbridge-indexer#139]
turns out with certain product configurations in the line can yield empty options like:
```
0 =>
  array (
  ),
  1 =>
  array (
    'value' => '43',
    'label' => 'Red',
    'sort_order' => 2,
  ),
)
```
changing the foreach (adding continue on empty value index) to 
```
foreach ($options as $option) {
                if(empty($option['value'])){
                        continue;
                }
                $values[] = (int)$option['value'];
                $productAttribute['values'][] = [
                    'value_index' => $option['value'],
                    'label' => $option['label'],
                ];
            }

```
fixes the issues and allows to reindex correctly

**Comment by janmyszkier** [comment]
summoning the powers of @afirlejczyk to do a CR and merge it so we can continue the works on the store :)

**Comment by afirlejczyk** [comment]
Hi @janmyszkier 

> turns out with certain product configurations in the line can yield empty options like.

Could you give an example ? Can you reproduce this error on clean Magento ?
I will merge fix tomorrow, but I am just curious.

Please update changelog.md, thanks!

**Comment by afirlejczyk** [comment]
Hi again!
I made a little different fix, and skip empty option in `GetConfigurableOptions`
https://github.com/DivanteLtd/magento2-vsbridge-indexer/pull/141

**Comment by janmyszkier** [comment]
@afirlejczyk but I gave you the outcome. Or did you mean "How do I configure magento product to get empty option like in the example?"

I didn't try YOUR change so cannot confirm it works, though.

**Comment by afirlejczyk** [comment]
Yes: "How do I configure magento product to get empty option like in the example?".

I am working on clean magento with sampla data, and didn't have this problem. 

nevertheless I merged a fix to develop.

**Comment by simonmaass** [comment]
have the same error on 1.5.0... will update

**Comment by afirlejczyk** [comment]
Please do, fix is available in 1.5.1
https://github.com/DivanteLtd/magento2-vsbridge-indexer/releases/tag/1.5.1

**Comment by simonmaass** [comment]
@afirlejczyk can confim - works!
